From bd87075f9455a329ed980bf65844749f042354fc Mon Sep 17 00:00:00 2001
From: Will Thompson <wjt@endlessos.org>
Date: Mon, 6 Feb 2023 10:00:05 +0000
Subject: [PATCH] Adjust Appstream ID to comply with D-Bus and Flatpak naming
 rules
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic
says:

> […] to ensure interoperability of the AppStream ID with other tools
> such as D-Bus […] it is […] strongly discouraged to start any
> segment of the ID with a digit.
> […]
> To ensures the greatest possible compatibility of an AppStream ID, it
> is recommended to […] prefix every leading digit of a section with an
> underscore […].  Since the underscore is not a valid character in domain
> names, the uniqueness of the ID is kept.

Flatpak follows D-Bus's restriction here.
---
 gobby-0.5.metainfo.xml.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gobby-0.5.metainfo.xml.in b/gobby-0.5.metainfo.xml.in
index 852836c..1372e4a 100644
--- a/gobby-0.5.metainfo.xml.in
+++ b/gobby-0.5.metainfo.xml.in
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop-application">
-  <id>de.0x539.gobby</id>
+  <id>de._0x539.gobby</id>
   <launchable type="desktop-id">gobby-0.5.desktop</launchable>
   <project_license>ISC</project_license>
   <metadata_license>CC0-1.0</metadata_license>
   <provides>
     <binary>gobby-0.5</binary>
+    <id>de.0x539.gobby</id>
   </provides>
   <_name>Gobby</_name>
   <_summary>Edit text files collaboratively</_summary>
-- 
2.30.2

